IPython/core/displayhook.py

```diff
@@ -168,6 +174,10 @@ def write_format_data(self, format_dict, md_dict=None):
         md_dict : dict (optional)
             The metadata dict to be associated with the display data.
         """
+        image_data = format_dict.get("image/png")
+        if image_data is not None and self.view_image(image_data):
+            return
+
```

Comment:
Hum that seem to be wrong, you likely do not want to modify anything in `IPython.core` as it will affect kernels regardless of the frontend. You likely want to get the data in the frontend and display it. It might be hard to find I'll see if I can have shot I finding where to put this.